x-pack/plugins/security_solution/scripts/endpoint/common/vm_services.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ToolingLog } from '@kbn/tooling-log';
+import execa from 'execa';
+import chalk from 'chalk';
+import { createToolingLogger } from '../../../common/endpoint/data_loaders/utils';
+import type { HostVm, HostVmExecResponse, SupportedVmManager } from './types';
+
+interface BaseVmCreateOptions {
+  name: string;
+  /** Number of CPUs */
+  cpus?: number;
+  /** Disk size */
+  disk?: string;
+  /** Amount of memory */
+  memory?: string;
+}
+
+interface CreateVmOptions extends BaseVmCreateOptions {
+  /** The type of VM manager to use when creating the VM host */
+  type: SupportedVmManager;
+  log?: ToolingLog;
+}
+
+/**
+ * Creates a new VM
+ */
+export const createVm = async ({ type, ...options }: CreateVmOptions): Promise<HostVm> => {
```

Comment:
FYI: eventually I will try to refactor all instances of code that access the VM managers directly to instead use the `HostVm` interface, which will allows us to abstract away from our tooling whatever the VM manager used is. IT will also allow us to introduce new VM manager and not impact our tooling